File: src/main/java/com/learnspark/plan/repository/ProjectRepository.java
package com.learnspark.plan.repository;

import com.learnspark.plan.entity.Project;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

/**
 * 学习项目仓库，所有查询自动过滤已软删除记录（通过实体上的 @SQLRestriction）。
 */
@Repository
public interface ProjectRepository extends JpaRepository<Project, String> {

    /** 获取用户的所有项目（自动排除已删除） */
    List<Project> findByUserIdOrderByCreatedAtDesc(String userId);

    /** 按 ID 和 userId 查询，确保用户只能访问自己的项目 */
    Optional<Project> findByIdAndUserId(String id, String userId);

    /** 判断项目是否属于指定用户（用于统计接口的归属校验，轻量查询） */
    boolean existsByIdAndUserId(String id, String userId);
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/repository/ProjectRepository.kt
package com.learnspark.server.domain.repository

import com.learnspark.server.domain.entity.Project
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface ProjectRepository : JpaRepository<Project, String> {
    fun findByUserId(userId: String): List<Project>
    fun findByUserIdOrderByUpdatedAtDesc(userId: String, pageable: Pageable): Page<Project>
    fun existsByIdAndUserId(id: String, userId: String): Boolean
}
